ui/hooks/gasFeeInput/useGasEstimates.js

```diff
@@ -111,7 +111,7 @@ export function useGasEstimates({
   const maximumCostInHexWei = getMaximumGasTotalInHexWei(gasSettings);
 
   if (editGasMode === EDIT_GAS_MODES.SWAPS) {
-    gasSettings = { ...gasSettings, gasLimit: decimalToHex(minimumGasLimit) };
+    gasSettings = { ...gasSettings, gasLimit: minimumGasLimit };
```

Comment:
For gas limit we usually use decimal value but seems minimumGasLimit in case of this swaps is an exception.
@dan437  can we be sure that swaps will always provide a hex value (and we never need to use default which is decimal).

It would I think be better to change [this](https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/swaps/view-quote/view-quote.js#L695) code to pass decimal.

Reply:
However, here we default it to hex https://github.com/MetaMask/metamask-extension/blob/develop/ui/hooks/gasFeeInput/useGasFeeInputs.js#L94

Reply:
And here we default it to a hex value as well https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/app/edit-gas-popover/edit-gas-popover.component.js#L47

Reply:
We definitely treat it as a decimal when rendering in the UI (e.g. in the `EditGasDisplay`), but we have a separate `minimumGasLimitDec` variable that we use for that

Reply:
Might be good to verify that all `minimumGasLimit` passed to `useGasFeeEstimates` are hex

Reply:
Good point @danjm , I will spend some time on this.